pkg/tecdsa/dkg/dkg.go

```diff
@@ -85,6 +86,101 @@ func (e *Executor) Execute(
 	return finalizationState.result(), endBlockNumber, nil
 }
 
+// StartedDKGEvent represents a DKG start event.
+type StartedDKGEvent struct {
+	Seed        *big.Int
+	BlockNumber uint64
+}
+
+// ResultSubmissionEvent represents a DKG result submission event. It is emitted
+// after a submitted DKG result is positively validated on the chain. It contains
+// the index of the member who submitted the result and a final public key of
+// the group.
+type ResultSubmissionEvent struct {
+	MemberIndex         uint32
+	GroupPublicKeyBytes []byte
+	Misbehaved          []uint8
+
+	BlockNumber uint64
+}
+
+type ResultSubmitter interface {
+	// SubmitDKGResult sends DKG result to a chain, along with signatures over
+	// result hash from group participants supporting the result.
+	// Signatures over DKG result hash are collected in a map keyed by signer's
+	// member index.
+	SubmitDKGResult(
+		participantIndex group.MemberIndex,
+		result *Result,
+		signatures map[group.MemberIndex][]byte,
+	) error
+	// IsGroupRegistered checks if group with the given public key is registered
+	// on-chain.
+	IsGroupRegistered(groupPublicKey []byte) (bool, error)
+	// OnDKGResultSubmitted registers a callback that is invoked when an on-chain
+	// notification of a new, valid submitted result is seen.
+	OnDKGResultSubmitted(
+		func(event *ResultSubmissionEvent),
+	) subscription.EventSubscription
```

Comment:
I think the only method we need here is:
```
SubmitResult(result *Result, memberIndex group.MemberIndex, signatures map[group.MemberIndex][]byte) error
```
We should implement that interface in `pkg/tbtc`. That implementation should be responsible for submitter eligibility as well.

Reply:
Done in e3e4bcd23c969076d0a7a55673f4c1dbc8dd38a3. although the interface looks a bit different.